src/components/Onboarding/onboarding-steps.tsx

```diff
@@ -1,15 +1,34 @@
 import { Stack, Stepper } from '@mantine/core';
-import { useState } from 'react';
+import { useEffect, useState } from 'react';
 
 import { StepCreateAccount } from './step-create-account';
 import { StepOnboardingFinished } from './step-onboarding-finished';
 import { StepUpdatePathMappings } from './step-update-path-mappings';
+import { api } from '~/utils/api';
 
 export const OnboardingSteps = ({ isUpdate }: { isUpdate: boolean }) => {
+  const maximumSteps = isUpdate ? 3 : 2;
+
   const [currentStep, setCurrentStep] = useState(0);
-  const nextStep = () => setCurrentStep((current) => (current < 3 ? current + 1 : current));
+
+  const nextStep = () => setCurrentStep((current) => (current < maximumSteps ? current + 1 : current));
+
   const prevStep = () => setCurrentStep((current) => (current > 0 ? current - 1 : current));
 
+  const { mutate: mutateConfigSchemaVersion } = api.config.updateConfigurationSchemaToLatest.useMutation();
+
+  const onFinishOnboarding = () => {
+    mutateConfigSchemaVersion();
+  };
+
+  useEffect(() => {
```

Comment:
Don't use useEffect for actions, it produces unpredictable bugs, use it in a event listener instead